### PR TITLE
[feat][pip] Refactor `SystemTopicBasedTopicPoliciesService` by TableView

### PR DESCRIPTION
**Pull request by mattisonchao**
### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Comment by mattisonchao** [comment]
https://lists.apache.org/thread/9v00sfpfxjpm775vgltkjoxwnllsgskg

**Comment by codelipenghui** [comment]
@mattisonchao I have moved this proposal to 3.2.0 since the 3.1.0 release process will start in a week.

File: pip/pip-284.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+## In Scope
+
+- Table view can help reduce complex logic in the previous implementation. 
+  - Demo of [new implementation](https://github.com/apache/pulsar/pull/20811/files#diff-76c55144efe73cecffaa74152c79271ed84f133bad6e4c2608633c35a674c5ebR60).
+  - Code of [old implementation](https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java).
+
+# Backward & Forward Compatibility
+
+There are no backwards compatibility issues because we use the same topic as before.
```
Comment (Technoboy-) [inline pip/pip-284.md:32]: Need to add the discussion mail to this PIP

**Comment by asafm** [comment]
@codelipenghui Don't forget proper title name. Instructions: https://github.com/apache/pulsar/tree/master/pip

**Pull request closed without merging**